sympy/solvers/recurr.py

```diff
@@ -822,3 +822,126 @@ def rsolve(f, y, init=None):
             solution = solution.subs(result)
 
     return solution
+
+
+def rfind(seq):
+    """
+    Given a list of expressions generated by an unknown function over
+    an integer parameter
+
+    .. math::
+
+        a_{0\le n\le N}=\{a_0, a_1, \ldots, a_N\},
+
+    it is possible to find recurrent relations between the values, and
+    further to deduce an expression producing the given values
+    (a "closed form"). At the moment SymPy can guess linear relations,
+    leading to linear recurrences, from which all expressions of type
+
+    .. math::
+
+        p_1\cdot c_1^n + p_2\cdot c_2^n + \cdots
+
+    (`p_i` polynomial in `n`, `c_i` constant expression) can be found,
+    provided enough values are given. Such sequences of values are called
+    "C-finite".
+
+    References
+    ==========
+
+    .. [1] Zeilberger, Doron. "The C-finite ansatz." The Ramanujan
+        Journal 31.1-2 (2013): 23-32.
+
+    See Also
+    ========
+    guess_seq
+
+    >>> from sympy.solvers.recurr import rfind
```

Comment:
I think, this should go "Examples" section.  And you should move it above (before References).